mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -684,7 +724,8 @@ impl SaplingSyncLoopHandle {
         rpc: &mut (dyn ZRpcOps + Send),
     ) -> Result<(), MmError<UpdateBlocksCacheErr>> {
         let current_block = rpc.get_block_height().await?;
-        let current_block_in_db = self.blocks_db.get_latest_block().await?;
+        let block_db = self.blocks_db.clone();
```

Comment:
do we really need this clone?

Reply:
yes, because mutable borrows was used here 
```rust
 if current_block >= from_block {
            let scan_blocks = rpc.scan_blocks(from_block, current_block, &block_db, self);
            scan_blocks.await?;
        }
```

Reply:
Well, I would say this is not a good design and `update_blocks_cache` shouldn't be part of `SaplingSyncLoopHandle` implementation.

Reply:
Ok, I'll improve it ..not sure why it was done that way before